### PR TITLE
Add support for trace with async logging

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes: #196.

### CODE REVIEW

File: spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -10,8 +10,13 @@
...
 	</appender>
 
+	<appender name="ASYNC" class="ch.qos.logback.classic.AsyncAppender">
+		<appender-ref ref="STACKDRIVER" />
+	</appender>
+
 	<root level="INFO">
 		<appender-ref ref="STDOUT" />
-		<appender-ref ref="STACKDRIVER" />
+<!--		<appender-ref ref="STACKDRIVER" />-->
```
Comment (elefeint) [inline spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample/src/test/resources/logback-test.xml:19]: Meant to comment out or remove?

**Review by elefeint — APPROVED** [review]

File: spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample/src/test/resources/logback-test.xml (hunk 1 above)
Comment (meltsufin) [reply spring-cloud-gcp-samples/spring-cloud-gcp-trace-sample/src/test/resources/logback-test.xml]: Sorry, wasn't really ready for review. Needed tests.

File: spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/TraceIdLoggingEnhancer.java [hunk 2]
```diff
@@ -117,4 +117,8 @@ public void enhanceLogEntry(LogEntry.Builder builder, ILoggingEvent e) {
 			builder.setSpanId(spanId);
 		}
 	}
+
+	public void setProjectIdProvider(GcpProjectIdProvider projectIdProvider) {
```
Comment (elefeint) [inline spring-cloud-gcp-logging/src/main/java/com/google/cloud/spring/logging/TraceIdLoggingEnhancer.java:121]: If only for tests, package private?

File: spring-cloud-gcp-logging/src/test/java/com/google/cloud/spring/logging/TraceIdLoggingEnhancerTests.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	TraceIdLoggingEnhancer enhancer = new TraceIdLoggingEnhancer();
+
+	@Before
+	public void before() {
+		enhancer.setProjectIdProvider(new GcpProjectIdProvider() {
+			@Override
+			public String getProjectId() {
+				return "gcp-project";
+			}
+		});
```
Comment (elefeint) [inline spring-cloud-gcp-logging/src/test/java/com/google/cloud/spring/logging/TraceIdLoggingEnhancerTests.java:49]: `() -> "gcp-project"`?

File: spring-cloud-gcp-logging/src/test/java/com/google/cloud/spring/logging/TraceIdLoggingEnhancerTests.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+		LogEntry.Builder logEntryBuilder = LogEntry.newBuilder(null);
+
+		ILoggingEvent mockLoggingEvent = mock(ILoggingEvent.class);
+		Map<String, String> map = new HashMap<>();
+		map.put(StackdriverTraceConstants.MDC_FIELD_TRACE_ID, "tid123");
+		map.put(StackdriverTraceConstants.MDC_FIELD_SPAN_ID, "sid123");
+		when(mockLoggingEvent.getMDCPropertyMap()).thenReturn(map);
+
+		MDC.put(StackdriverTraceConstants.MDC_FIELD_TRACE_ID, "tid-mdc");
+		MDC.put(StackdriverTraceConstants.MDC_FIELD_SPAN_ID, "sid-mdc");
```
Comment (elefeint) [inline spring-cloud-gcp-logging/src/test/java/com/google/cloud/spring/logging/TraceIdLoggingEnhancerTests.java:75]: Should MDC threadlocal get cleared at the end of the test? Curious that it did not mess with the test below it, which needs MDC threadlocal unset.

**Review by elefeint — APPROVED** [review]
Who would be calling `TraceIdLoggingEnhancer.setCurrentTraceId()`? Should this be documented?

**Pull request merged**